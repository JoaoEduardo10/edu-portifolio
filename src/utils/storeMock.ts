import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const store = mockStore({
  menu: {
    menuOn: false,
    colorOn: true,
  },
  completeProject: {
    id: "1234",
    project: false,
  },
});

export { store };
