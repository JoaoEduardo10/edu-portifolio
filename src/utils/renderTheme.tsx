/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../style/themes";
import { Provider } from "react-redux";
import { GlobalStyles } from "../style/globalStyles";
import { ToolkitStore } from "@reduxjs/toolkit/dist/configureStore";
import { store } from "./storeMock";

const renderTheme = (children: React.ReactNode, storeParams?: ToolkitStore) => {
  const isStore = storeParams ?? store;
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={isStore}>
        {children}
        <GlobalStyles />
      </Provider>
    </ThemeProvider>
  );
};

export { renderTheme };
