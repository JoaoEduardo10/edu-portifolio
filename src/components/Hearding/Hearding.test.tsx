import React from "react";
import { screen } from "@testing-library/react";
import { Hearding } from ".";
import { renderTheme } from "../../utils/renderTheme";
import { describe, it } from "vitest";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

describe("<Hearding />", () => {
  const store = mockStore({
    menu: {
      menuOn: false,
      colorOn: false,
    },
    completeProject: {
      id: "1234",
      project: false,
    },
  });

  it("should a render Heading with color white", () => {
    renderTheme(<Hearding text="Um test" />);

    const heading = screen.getByRole("heading", { name: "Um test" });

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveStyle({
      color: "#fff",
    });
    expect(heading).toMatchSnapshot();
  });

  it("should a render Heading with color Black", () => {
    renderTheme(<Hearding text="Um test" />, store);

    const heading = screen.getByRole("heading", { name: "Um test" });

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveStyle({
      color: "#000",
    });
    expect(heading).toMatchSnapshot();
  });
});
