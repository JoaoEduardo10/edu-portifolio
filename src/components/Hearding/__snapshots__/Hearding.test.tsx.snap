// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Hearding /> > should a render Heading with color Black 1`] = `
.c0 {
  text-align: center;
  font-size: 4rem;
  margin-bottom: 2.5rem;
  color: #000;
}

<h2
  class="c0"
>
  Um test
</h2>
`;

exports[`<Hearding /> > should a render Heading with color white 1`] = `
.c0 {
  text-align: center;
  font-size: 4rem;
  margin-bottom: 2.5rem;
  color: #fff;
}

<h2
  class="c0"
>
  Um test
</h2>
`;
